rclcpp/src/rclcpp/parameter_client.cpp

```diff
@@ -356,6 +356,28 @@ SyncParametersClient::SyncParametersClient(
     std::make_shared<AsyncParametersClient>(node, remote_node_name, qos_profile);
 }
 
+SyncParametersClient::SyncParametersClient(
+  rclcpp::Node * node,
+  const std::string & remote_node_name,
+  const rmw_qos_profile_t & qos_profile)
+: SyncParametersClient(
+    std::make_shared<rclcpp::executors::SingleThreadedExecutor>(),
+    node,
+    remote_node_name,
+    qos_profile)
+{}
+
+SyncParametersClient::SyncParametersClient(
+  rclcpp::executor::Executor::SharedPtr executor,
+  rclcpp::Node * node,
+  const std::string & remote_node_name,
+  const rmw_qos_profile_t & qos_profile)
+: executor_(executor), node_(node)
```

Comment:
This is dangerous, because you're creating a shared pointer to Node (in `node_`) from a pointer that is already owned by another shared pointer (`node` is the pointer and all nodes are created as shared pointers to support `std::enable_shared_from_this`).

The SyncParametersClient needs to be changed to hold a raw pointer (optionally or always) rather than always having a shared pointer to node which cannot be safely fulfilled with a raw pointer as input.

Reply:
I changed the SyncParametersClient member node_ to a raw pointer.
SyncParametersClient has still constructors which accept a raw or shared pointer to a node.